### ISSUE TITLE
Better CI with integration tests

### ISSUE THREAD

**Issue by jkozdon** [CliMA/ClimateMachine.jl#652]
Right now we just run a few time steps of the atmos model for integration tests (dycoms, RTB, etc.), but it is a little concerning that the CI didn't break with the changes from #648

**Comment by jkozdon** [comment]
Currently the [Dycoms] CI passes, but that just means that the code [didn't crash](https://github.com/climate-machine/CLIMA/blob/461657afa53ad608015a27f7c91dd8c15e605810/experiments/LES/dycoms.jl#L154-L185) since there is no `@test` in the test file.

_Originally posted by @jkozdon in https://github.com/climate-machine/CLIMA/pull/660#issuecomment-581980491_

Right. I'm hoping @akshaysridhar or @smarras79 can suggest a useful `@test` for the current 100sec simulation at some point.

_Originally posted by @kpamnany in https://github.com/climate-machine/CLIMA/pull/660#issuecomment-581981548_

> Right. I'm hoping @akshaysridhar or @smarras79 can suggest a useful `@test` for the current 100sec simulation at some point.

We need something to compare against when we perform the time-integrator study to make sure we know the stability condition parameters for the inviscid and viscous processes. Having a working DYCOMS up to the point that the stability measures (e.g., horizontal and vertical Courant numbers and Peclet numbers) equilibrate will be helpful.

_Originally posted by @fxgiraldo in https://github.com/climate-machine/CLIMA/pull/660#issuecomment-581996076_

> > Right. I'm hoping @akshaysridhar or @smarras79 can suggest a useful `@test` for the current 100sec simulation at some point.
> 
> We need something to compare against when we perform the time-integrator study to make sure we know the stability condition parameters for the inviscid and viscous processes. Having a working DYCOMS up to the point that the stability measures (e.g., horizontal and vertical Courant numbers and Peclet numbers) equilibrate will be helpful.

Since we now have a way to I/O restart files, one thing that could be good to test only 100s of DYCOMS is to start the test from a restart file in the middle of the simulation so that all the initial transient is gone. Run it for 100s, and look at the norms difference for that slab of the run.
@akshaysridhar what do you think of this option?

_Originally posted by @smarras79 in https://github.com/climate-machine/CLIMA/pull/660#issuecomment-582001464_

**Comment by blallen** [comment]
Can someone link me to the way to restart files?

This will help us implement tests for the ocean model beyond checking that we preserve incompressibility.

**Comment by kpamnany** [comment]
The PR that demonstrates restart capability is [here](https://github.com/climate-machine/CLIMA/pull/576) but it needs to be refactored and also enhanced to work properly with multiple MPI ranks. The other issue is the restart files. We probably don't want to commit large data files to the repo, so will have to use something like `Datadeps.jl`. @charleskawczynski is exploring this in #642, but that's a little stuck at the moment because of package upgrade issues.

In short, it's not worked out yet.

**Comment by kpamnany** [comment]
Restart support was added in #850.

**Comment by jkozdon** [comment]
Possibly using https://github.com/CliMA/ClimateMachine.jl/pull/1118 (Add quick debug/regression statistics on raw state) would be a more reliable way for integration tests that the L2 integral (which can be insensitive to changes).

**Comment by jkozdon** [comment]
> Possibly using #1118 (Add quick debug/regression statistics on raw state) would be a more reliable way for integration tests that the L2 integral (which can be insensitive to changes).

See:

- [`experiments/OceanBoxGCM/ocean_gyre.jl`](https://github.com/CliMA/ClimateMachine.jl/blob/258ad947d71e9ff9fbeb14bbf41e8ab59fb73d29/experiments/OceanBoxGCM/ocean_gyre.jl)
- [`experiments/OceanBoxGCM/ocean_gyre_refvals.jl`](https://github.com/CliMA/ClimateMachine.jl/blob/258ad947d71e9ff9fbeb14bbf41e8ab59fb73d29/experiments/OceanBoxGCM/ocean_gyre_refvals.jl)

h/t: @blallen

**Comment by blallen** [comment]
> > Possibly using #1118 (Add quick debug/regression statistics on raw state) would be a more reliable way for integration tests that the L2 integral (which can be insensitive to changes).
> 
> See:
> 
> * [`experiments/OceanBoxGCM/ocean_gyre.jl`](https://github.com/CliMA/ClimateMachine.jl/blob/258ad947d71e9ff9fbeb14bbf41e8ab59fb73d29/experiments/OceanBoxGCM/ocean_gyre.jl)
> * [`experiments/OceanBoxGCM/ocean_gyre_refvals.jl`](https://github.com/CliMA/ClimateMachine.jl/blob/258ad947d71e9ff9fbeb14bbf41e8ab59fb73d29/experiments/OceanBoxGCM/ocean_gyre_refvals.jl)
> 
> h/t: @blallen

also h/t: @christophernhill, he added all of that

**Comment by christophernhill** [comment]
@jkozdon feel free to ping if I can help.
